pyaedt/modeler/cad/Primitives.py

```diff
@@ -8215,3 +8242,553 @@ def convert_segments_to_line(self, object_name):
                 ]
             )
         return True
+
+
+class PrimitivesBuilder(object):
+    """Create primitives based on json file or dictionary of properties.
+
+    Parameters
+    ----------
+    app :
+        Inherited parent object.
+    input_file : str, optional
+        Path to a json file containing primitive settings.
+    input_dict : dict, optional
+        Dictionary containing primitive settings.
+
+    Returns
+    -------
+    :class:`pyaedt.modeler.cad.PrimitivesBuilder`
+        Primitives builder object if succeeded.
+
+    Examples
+    --------
+    >>> from pyaedt import Hfss
+    >>> from pyaedt.modeler.cad.Primitives import PrimitivesBuilder
+    >>> aedtapp = Hfss()
+    >>> primitive_file = "primitive_file.json"
+    >>> primitives_builder = PrimitivesBuilder(aedtapp, input_file=primitive_file)
+    >>> primitives_builder.create()
+    >>> aedtapp.release_desktop()
+    """
+
+    def __init__(self, app, input_file=None, input_dict=None):
+        self._app = app
+        props = {}
+        if not input_dict and not input_file:  # pragma: no cover
+            msg = "Either one of a json file or a dictionary has to be passed as input."
+            self.logger.error(msg)
+            raise TypeError(msg)
+        elif input_file:
+            file_format = os.path.splitext(os.path.basename(input_file))[1]
+            if file_format == ".json":
+                props = json_to_dict(input_file)
+            elif file_format == ".csv":
+                import re
+
+                from pyaedt.generic.general_methods import read_csv_pandas
+
+                csv_data = read_csv_pandas(filename=input_file)
+                primitive_type = csv_data.columns[0]
+                primitive_type_cleaned = re.sub(r"^#\s*", "", primitive_type)
+
+                if primitive_type_cleaned in ["Blocks Cylinder", "Cylinder"]:
+                    props = self._read_csv_cylinder_props(csv_data)
+                if primitive_type_cleaned in ["Blocks Prism", "Prism"]:
+                    props = self._read_csv_prism_props(csv_data)
+                if not props:
+                    msg = "CSV file not valid."
+                    self.logger.error(msg)
+                    raise TypeError(msg)
+            else:
+                msg = "Format not valid."
```

Comment:
```suggestion
                msg = "Format is not valid."
```
A message must have be a complete sentence (so have a verb).